ffs/pb/fps.proto

```diff
@@ -6,92 +6,227 @@ option java_package = "io.textile.filecoin.ffs.pb";
+}
+
+message WalletAddrRequest {
+}
+
+message WalletAddrReply {
+   string addr = 1;
+}
+
+message GetDefaultCidConfigRequest {
+   string cid = 1;
+}
+
+message GetDefaultCidConfigReply {
+   CidConfig config = 1;
+}
+
+message GetCidConfigRequest {
+   string cid = 1;
+}
+
+message GetCidConfigReply {
+   CidConfig config = 1;
+}
+
+message SetDefaultCidConfigRequest {
+   DefaultCidConfig config = 1;
+}
+
+message SetDefaultCidConfigReply {
 }
 
 message ShowRequest {
    string cid = 1;
 }
 
+message ShowReply {
+   CidInfo cidInfo = 1;
+}
+
 message InfoRequest {
 }
 
 message InfoReply {
-   string id = 1;
-   WalletInfo wallet = 2;
-   repeated string pins = 3;
+   InstanceInfo info = 1;
 }
 
-message WalletInfo {
-   string address = 1;
-   uint64 balance = 2;
+message WatchRequest {
+   repeated string jids = 1;
 }
 
-message ShowReply {
+message WatchReply {
+   Job job = 1;
+}
+
+message PushConfigRequest {
    string cid = 1;
-   int64 created = 2;
-   HotInfo hot = 3;
-   ColdInfo cold = 4;
-   
-   message HotInfo {
-      int64 size = 1;
-      IpfsHotInfo ipfs = 2;
-   }
-
-   message IpfsHotInfo {
-      int64 created = 1;
-   }
-
-   message ColdInfo {
-      FilInfo filecoin = 1;
-   }
-
-   message FilInfo {
-      string dataCid = 1;
-      repeated FilStorage proposals = 2;
-   }
-
-   message FilStorage {
-      string proposalCid = 1;
-      int64 duration = 2;
-      int64 activationEpoch = 3;
-   }
+   CidConfig config = 2;
+   bool override = 3;
+}
+
+message PushConfigReply {
+   string jobID = 1;
+}
+
+message GetRequest {
+    string cid = 1;
+}
+
+message GetReply {
+    bytes chunk = 1;
 }
 
+message CloseRequest {
+}
+
+message CloseReply {
+}
+
+message AddToHotRequest {
+  bytes chunk = 1;
+}
+
+message AddToHotReply {
+  string cid = 1;
+}
+
+
+
 service API {
-    rpc AddCid(AddCidRequest) returns (AddCidReply) {}
-    rpc AddFile(stream AddFileRequest) returns (AddFileReply) {}
-    rpc Get(GetRequest) returns (stream GetReply) {}
-    rpc Create(CreateRequest) returns (CreateReply) {}
-    rpc Info(InfoRequest) returns (InfoReply) {}
-    rpc Show(ShowRequest) returns (ShowReply) {}
+   rpc Create(CreateRequest) returns (CreateReply) {}
+   rpc ID(IDRequest) returns (IDReply) {}
+   rpc WalletAddr(WalletAddrRequest) returns (WalletAddrReply) {}
+   rpc GetDefaultCidConfig(GetDefaultCidConfigRequest) returns (GetDefaultCidConfigReply) {}
+   rpc GetCidConfig(GetCidConfigRequest) returns (GetCidConfigReply) {}
+   rpc SetDefaultCidConfig(SetDefaultCidConfigRequest) returns (SetDefaultCidConfigReply) {}
+   rpc Show(ShowRequest) returns (ShowReply) {}
+   rpc Info(InfoRequest) returns (InfoReply) {}
+   rpc Watch(WatchRequest) returns (stream WatchReply) {}
+   rpc PushConfig(PushConfigRequest) returns (PushConfigReply) {}
+   rpc Get(GetRequest) returns (stream GetReply) {}
+   rpc Close(CloseRequest) returns (CloseReply) {}
+   rpc AddToHot(stream AddToHotRequest) returns (AddToHotReply) {}
```

Comment:
`AddFile` is now `AddHot` and it only adds the file to IPFS and returns the CID that can be used in `PushConfig`. Figured its better to keep it simple. If any user of the client (eg the CLI) wants to created more automated UX, they can.